pandas/core/generic.py

```diff
@@ -6089,20 +6089,22 @@ def fillna(
                 )
 
             elif isinstance(value, (dict, ABCSeries)):
+                new_data = self if inplace else self.copy()
+
                 if axis == 1:
-                    raise NotImplementedError(
-                        "Currently only can fill "
-                        "with dict/Series column "
-                        "by column"
-                    )
+                    for i in range(len(new_data.columns)):
```

Comment:
Can you continue using `new_data.items()` in both branches here? Can wrap in `enumerate` to get the index

Reply:
@WillAyd 
need `iloc` method. "fillna" using series from each columns by `.items` not guarantee filling NA if there is duplicated columns.

Reply:
Makes sense. So can you still use `items` but wrap in enumerate? You can then use `.iloc` on the left hand side for assignment